### PR TITLE
fix: Monitors dissapearing from Unit upon edit

### PR DESCRIPTION
**Pull request by KevLehman**
## Proposed changes (including videos or screenshots)

Introduced here: #30734

## Issue(s)

https://rocketchat.atlassian.net/browse/SUP-572
## Further comments

Fixed an issue that caused the UI to send an invalid _id for monitors that were already saved on the unit. This caused the BE to ignore these monitors (as the _id didn't exist) and save only the newly added items to the list (if any)

This was caused because on an old refactor, the destructuring of the `monitors` property changed from `monitorId` (on JS) to `_id` (in TS)

As the elements were coming from the `unit_monitors` collection instead of the `users` collection, here `_id` represented something totally different. When the BE eliminated the elements that were not valid monitors from the array, these were excluded (which is correct) and not saved.

### CODE REVIEW

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts [hunk 1]
```diff
@@ -117,6 +122,24 @@ test.describe('OC - Manage Units', () => {
...
+			await expect(poOmnichannelUnits.inputSearch).toBeVisible();
+			await poOmnichannelUnits.search(editedUnitName);
+			await expect(poOmnichannelUnits.findRowByName(editedUnitName)).toBeVisible();
+			await poOmnichannelUnits.findRowByName(editedUnitName).click();
+			await expect(poOmnichannelUnits.contextualBar).toBeVisible();
+
+			await expect(poOmnichannelUnits.inputMonitors).toHaveText(/user2/);
+			await expect(poOmnichannelUnits.inputMonitors).toHaveText(/user3/);
+		});
+
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:142]: Can we add a new test case removing one user from the list (of 2), save it, and see if it behaves correctly?

**Review by MarcosSpessatto — DISMISSED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:1]: this amount of assertion checking if the elements are visible is really necessary?

eg: checking if the contextualBar is visible. Theres a step to to click in a button on contextualBar, the test you fail if its no visible, so what's the purpose of the assertion? I don't see a real value using this approach, kinda repetitive in my point of view

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts
Comment (KevLehman) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:1]: Good question. IDK. I copied the pattern from the tests above it and adapted my checks 😬

Comment (KevLehman) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-units.spec.ts:1]: BUT, removed the unneeded expects

**Review by dougfabris — DISMISSED** [review]
LGTM!

**Comment by KevLehman** [comment]
Hey @dougfabris i forgot to add a changeset before, mind to review this one again? 😬

**Review by dougfabris — APPROVED** [review]

**Review by MartinSchoeler — APPROVED** [review]

**Pull request merged**